controllers/chat/getSignedChannelList.js

```diff
@@ -0,0 +1,79 @@
+const {responseSuccess} = require('../../utils/Responses');
+const GetstreamSingleton = require('../../vendor/getstream/singleton');
+
+const MAX_ITERATIONS = 10;
+
+const __queryChannelBuilder = async (userId, limit, offset) => {
+  const client = GetstreamSingleton.getChatInstance();
+  const channels = await client.queryChannels({members: {$in: [userId]}}, [{last_message_at: -1}], {
+    limit: limit,
+    offset: offset,
+    state: true
+  });
+  await client.disconnectUser();
+  return channels;
+};
+
+const __transformChannelData = (channel) => {
+  const newChannel = {...channel.data};
+  delete newChannel.config;
+  delete newChannel.own_capabilities;
+
+  const members = [];
+  Object.keys(channel.state.members).forEach((member) => {
+    members.push(channel?.state?.members[member]);
+  });
+
+  return {
+    ...newChannel,
+    members,
+    unreadCount: channel.state.unreadCount,
+    messages: channel.state.messages
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `__transformChannelData` function is creating a new array `members` by iterating over `channel.state.members`. This can be simplified using `Object.values()`, which returns an array of a given object's own enumerable property values. Here's how you can do it:

```diff
-  const members = [];
-  Object.keys(channel.state.members).forEach((member) => {
-    members.push(channel?.state?.members[member]);
-  });
+  const members = Object.values(channel?.state?.members || {});
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->